src/v/raft/configuration_manager.cc

```diff
@@ -67,16 +67,35 @@ ss::future<> configuration_manager::prefix_truncate(model::offset offset) {
     vlog(_ctxlog.trace, "Prefix truncating configurations at {}", offset);
     return _lock.with([this, offset] {
         auto it = _configurations.lower_bound(offset);
```

Comment:
Again, why not 
```
auto it = _configurations.upper_bound(offset);
_configurations.erase(_configurations.begin(), std::prev(it));
```

this way there is always a configuration with the offset before valid log offsets and there are no corner cases in `offset_translator::delta`

Reply:
this would not work for the case where `upper_bound` is equal to `_configurations.begin()`, we would need special case for that. f.e. first configuration has offset 20, we prefix truncate at 10. It would require additional special case.